pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -108,7 +151,9 @@ func (s *NodeController) Run(period time.Duration, syncNodeList, syncNodeStatus
 			glog.Errorf("Error loading initial static nodes: %v", err)
 		}
 	}
+	nodes = s.DoChecks(nodes)
 	nodes, err = s.PopulateAddresses(nodes)
+	nodes = reconcilePodCIDRs(nodes, &api.NodeList{})
```

Comment:
The current implementation I'm proposing here makes an assumption that CIDR assignment is done at the point when you first see a Node (i.e. just before creating it in the apiserver). I thought it's a bit harder to have a logic that also looks for nodes that are in the apiserver but don't have the IP assigned (it's all a bit overcomplicated because we still need to support cloudproviders that do not have the possibility to reconfigure IP range on the fly...). I understand it might not be the best way to solve all this, happy to take any proposals.